resources/sources.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Author:: Blair Hamilton (bhamilton@draftkings.com>)
+# Author:: Jonathan Morley (morley.jonathan@gmail.com>)
+# Cookbook Name:: nuget
+# Resource:: source
+#
+# Copyright 2015, Blair Hamilton
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/mixin/shell_out'
+include Chef::Mixin::ShellOut
+
+property :name, kind_of: String, name_attribute: true
```

Comment:
The name resource is defined by chef itself so there's no need to define it here. What you probably want to create is a source_name property which would also be a name_property. That allows the name of the source to differ from the name the user gives the resource. You'd then use new_resource.source_name throughout the resource instead of new_resource.name.